### ISSUE TITLE
Information about the type of data structure from object.

### ISSUE THREAD

**Issue by 24sharkS** [randy3k/collections#14]
I am using `dict()` object in a function and want to perform check to make sure that the argument passed to the function is of dict type.

The class function doesn't provide this information.

```
d <- dict(list(a = 1,b = 2))  
class(d)
[1] "environment"
```
Any workaround for this issue?

**Comment by 24sharkS** [comment]
@randy3k is there any function for this purpose?

**Comment by randy3k** [comment]
That's a valid request. There is no such thing now.

The closest thing is
```
is.environment(d) && identical(parent.env(d), asNamespace("collections"))
```
But it only checks if the object is from `collections`, it doesn't check if it is a `dict` or other containers.

**Comment by 24sharkS** [comment]
In addition to above we can also do something like this, which will confirm the type too.

```
strsplit(capture.output(d$print)," ")[[1]][1] == "dict"
```

**Comment by randy3k** [comment]
closed by b308f8f
